src/workload/mod.rs

```diff
@@ -807,3 +820,45 @@ impl Ratelimit {
         limit
     }
 }
+
+fn estimate_random_bytes_needed(length: usize, compression_ratio: f64) -> usize {
+    // if compression ratio is low, all bytes should be random
+    if compression_ratio <= 1.0 {
```

Comment:
Do we want to short circuit exit if `length == 0` as well?

Reply:
I don't think it's needed. Initializing the PRNG shouldn't be too expensive, and it happens only once per keyspace/topic-space.